jina/clients/base/helper.py

```diff
@@ -103,7 +118,15 @@
         req_dict['exec_endpoint'] = req_dict['header']['exec_endpoint']
         if 'target_executor' in req_dict['header']:
             req_dict['target_executor'] = req_dict['header']['target_executor']
-        return await self.session.post(url=self.url, json=req_dict).__aenter__()
+        for retry in range(1, self.max_attempts + 1):
+            try:
+                return await self.session.post(url=self.url, json=req_dict).__aenter__()
+            except:
+                if retry == self.max_attempts:
+                    raise
+                else:
+                    wait_time = random.uniform(0, min(self.initial_backoff*self.backoff_multiplier**(retry-1), self.max_backoff))
```

Comment:
I might have missed the design discussion on this but why is the waiting time random here ?

Reply:
it is the definition of what is done in grpc